### PR TITLE
Pass ServerPlayerEntity instead of PlayerEntity to the drop item listener

### LINKED ISSUE #74: DropItemListener should give ServerPlayerEntity and not PlayerEntity

### PR DESCRIPTION
**Pull request by haykam821**
Fixes #74

### CODE REVIEW

File: src/main/java/xyz/nucleoid/plasmid/mixin/game/event/ScreenHandlerMixin.java [hunk 1]
```diff
@@ -64,9 +64,10 @@ private void close(PlayerEntity player, CallbackInfo ci) {
 
     private boolean shouldBlockThrowingItems(PlayerEntity player, int slot, ItemStack stack) {
         ManagedGameSpace gameSpace = ManagedGameSpace.forWorld(player.world);
-        if (gameSpace != null && gameSpace.containsPlayer((ServerPlayerEntity) player)) {
+		ServerPlayerEntity serverPlayer = (ServerPlayerEntity) player;
```
Comment (Gegy) [inline src/main/java/xyz/nucleoid/plasmid/mixin/game/event/ScreenHandlerMixin.java:67]:
```suggestion
	    ServerPlayerEntity serverPlayer = (ServerPlayerEntity) player;
```

Comment (haykam821) [reply src/main/java/xyz/nucleoid/plasmid/mixin/game/event/ScreenHandlerMixin.java]: Fixed in 907e319f91537b25c4a0a522d901d8c4fa82c4df

File: src/main/java/xyz/nucleoid/plasmid/mixin/game/event/ScreenHandlerMixin.java [hunk 2]
```diff
@@ -64,9 +64,10 @@ private void close(PlayerEntity player, CallbackInfo ci) {
 
     private boolean shouldBlockThrowingItems(PlayerEntity player, int slot, ItemStack stack) {
         ManagedGameSpace gameSpace = ManagedGameSpace.forWorld(player.world);
-        if (gameSpace != null && gameSpace.containsPlayer((ServerPlayerEntity) player)) {
+	    ServerPlayerEntity serverPlayer = (ServerPlayerEntity) player;
```
Comment (Gegy) [inline src/main/java/xyz/nucleoid/plasmid/mixin/game/event/ScreenHandlerMixin.java:67]: Hm, indentation seems odd here still

Comment (haykam821) [reply src/main/java/xyz/nucleoid/plasmid/mixin/game/event/ScreenHandlerMixin.java]: Fixed in defeb971c1d89c7330a7c0665a36bc1c8e405774

**Review by Gegy — APPROVED** [review]

**Pull request merged**